lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1661,8 +1660,14 @@ sub revivified_alignment_bam_file_paths {
         return @bams;
     }
 
-    my $revivified_bam = File::Spec->join($p{disk_allocation}->absolute_path, 'all_sequences.bam');
-    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
+    my $temp_allocation = $self->_get_temp_allocation($self->output_dir);
+    UR::Context->process->add_observer(
```

Comment:
What do you guys think of the pattern of using a destroyer as a wrapper? http://search.cpan.org/~adamk/Object-Destroyer-2.01/lib/Object/Destroyer.pm
I'm not sure how that would interact with UR, and it would add a new perl dependency.